src/test/java/com/zendesk/maxwell/row/RowMapTest.java

```diff
@@ -55,4 +56,198 @@ public void testTimestampConversion() throws Exception {
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis, pKeys, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("name", "example");
+
+		String jsonString = rowMap.pkToJson(RowMap.KeyFormat.ARRAY);
+
+		Assert.assertEquals("[\"MyDatabase\",\"MyTable\",[{\"id\":\"9001\"},{\"name\":\"example\"}]]",
+				jsonString);
+
+
+	}
+
+	@Test
+	public void testBuildPartitionKey() {
+		long timestampMillis = 1496712943447L;
+
+		List<String> pKeys = new ArrayList<>();
+
+		pKeys.add("id");
+
+		pKeys.add("first_name");
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis, pKeys, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+
+		List<String> partitionColumns = Arrays.asList("id", "first_name", "middle_name", "last_name", "salary", "department");
+
+		String partitionKey = rowMap.buildPartitionKey(partitionColumns, "primary_key");
+
+		Assert.assertEquals("9001foobuzzbar4000science", partitionKey);
+
+	}
+
+	@Test
+	public void testBuildPartitionKeyWithEmptyList() {
+		long timestampMillis = 1496712943447L;
+
+		List<String> pKeys = new ArrayList<>();
+
+		pKeys.add("id");
+
+		pKeys.add("first_name");
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis, pKeys, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+
+		List<String> partitionColumns = new ArrayList<>();
+
+		String partitionKey = rowMap.buildPartitionKey(partitionColumns, "primary_key");
+
+		Assert.assertEquals("9001foo", partitionKey);
+
+	}
+
+	@Test
+	public void testToJSONWithRawJSONData() throws IOException {
+		long timestampMillis = 1496712943487L;
+
+		Position position = new Position(new BinlogPosition(1L, "binlog-0001"), 0L);
+		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", timestampMillis,
+				null, position);
+
+		rowMap.putData("id", "9001");
+		rowMap.putData("first_name", "foo");
+		rowMap.putData("middle_name", "buzz");
+		rowMap.putData("last_name", "bar");
+		rowMap.putData("salary", "4000");
+		rowMap.putData("department", "science");
+        rowMap.putData("rawJSON", new RawJSONString("{\"database\":\"MyDatabase\",\"table\":\"User\",\"type\":\"insert\",\"ts\":1486439516,\"xid\":5694,\"commit\":true,\"data\":{\"UserID\":20,\"FirstName\":\"Fiz\",\"MiddleName\":null,\"LastName\":\"Buz\",\"Version\":1486703131000}}\n"));
```

Comment:
You can probably make this shorter - it's effectively string concat, right? It could probably be pulled out into a variable and reused, too.